templates/CRM/Block/FullTextSearch.tpl

```diff
@@ -33,18 +33,18 @@
       <option value="Contact">{ts}Contacts{/ts}</option>
 {/if}
       <option value="Activity">{ts}Activities{/ts}</option>
-{if call_user_func(array('CRM_Core_Permission','access'), 'CiviCase')}
+{crmPermission has='administer CiviCase'}
```

Comment:
"CRM_Core_Permission::access" and "CRM_Core_Permission::check" are very different functions.

Reply:
@demeritcowboy yes agree, but other then checking the enabled component the fn check for `administer $component` permission. In this patch I am considering if the system doesn't have particular component enabled say CiviCase then it won't have `administer CiviCase` permission. So I am simply making a permission check.